nemo/collections/speechlm2/parts/metrics/bleu.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import defaultdict
+
+import sacrebleu
+import torch
+from whisper_normalizer.english import EnglishTextNormalizer
+
+from nemo.utils import logging
+
+
+class BLEU:
```

Comment:
why to redefine this again and not use from asr collection? To be independent of other collections?

Reply:
This collection has a redesigned mechanism for multi-validation sets and the metrics are built with support for keeping information from multiple datasets. It's more convenient to have them available separately, at least while the collection is still under very active development. If it's possible to merge them later, we can do that.